engines/grim/emi/sound/emisound.cpp

```diff
@@ -663,8 +667,18 @@ void EMISound::initMusicTable() {
 		_musicTable = initMusicTableDemo("Music/FullMonkeyMap.imt");
 		_musicPrefix = "Music/";
 	} else if (g_grim->getGamePlatform() == Common::kPlatformPS2) {
-		_musicTable = emiPS2MusicTable;
-		_numMusicStates = ARRAYSIZE(emiPS2MusicTable);
+		static_assert(ARRAYSIZE(emiPS2MusicTable) == 126, "emiPS2MusicTable bad size");
```

Comment:
```suggestion
		STATIC_ASSERT(ARRAYSIZE(emiPS2MusicTable) == 126, emiPS2MusicTable_bad_size);
```

Reply:
Why? Other places use static_assert, too -- are they in error then, too?

Reply:
@fingolfin : Nice to see you again. I think the issue is that static_assert was introduced by C++-11 and we still support building with C++98 (for _most_ engines and backend code). Some engines which were imported are marked as C++-11 only as their source code heavily relies on C++-11 constructs and removing / changing this is problematic or has not been a priority i.e. AGS.

Reply:
The GRIM engine is not C++-11 only as per https://github.com/scummvm/scummvm/blob/master/engines/grim/configure.engine ... Contrast the AGS engine which has the cxx11 keyword: https://github.com/scummvm/scummvm/blob/master/engines/ags/configure.engine

Reply:
The STATIC_ASSERT macro is supported across all compilers with a fallback using a macro definition here:
https://github.com/scummvm/scummvm/blob/master/common/scummsys.h#L144

Reply:
Makes sense, thanks for the explanation

Reply:
Have audited the source code in master and the only case of static_assert usage was in the TWINE engine. Have removed with c9fcf5e96f6e122967f7b7b32a472e0c85329285. This should not cause a conflict with this as you don't touch TWINE with your changes.

Reply:
@digitall Won't your commit https://github.com/scummvm/scummvm/commit/c9fcf5e96f6e122967f7b7b32a472e0c85329285 break compilation with the fallback implementation of STATIC_ASSERT, as it uses a string literal as message, but the fallback requires the message to be turned into an identifier? Is ScummVM CI running a test compiling ScummVM without C++11 support?

Reply:
@fingolfin Did a quick check with --disable-c++11 passed to configure script and this still builds all engines fine... HOWEVER, that is partly as the TWINE engine is marked as C++-11 only: https://github.com/scummvm/scummvm/blob/master/engines/twine/configure.engine and thus is disabled.

This still builds fine with C+-11 enabled as the implementation of STATIC_ASSERT will default to using static_assert anyway.

I am not sure about that fallback, including the doxygen comment saying "underscore" delimited for the message string... but that does not affect this change specifically i.e. STATIC_ASSERT should be a replacement / wrapper for static_assert with same calling convention.

As for CI, best to talk to @lotharsm about that as I haven't done much with newer dockerized buildbot, though my build does show some warnings it is building fine.

Reply:
Thanks for the mention! AFAIK the majority of the platforms on the buildbot is running on C++-11 already, but I _think_ we have some non-C++-11 builds left due to older toolchains for the more exotic ports. @lephilousophe, do we have an overview of which platforms use C++-11 somewhere hidden in the buildbot config?

Reply:
We don't have an exhaustive list but I can already say that: 
- gp2x is not C++11 compatible at all (-std=c++11 not recognized)
- macosx-i386 has partial c++11 support (its libstdc++ is not c++11 compliant but compiler is, because it's modern clang)

I did several tests and filled them in the [Wiki](https://wiki.scummvm.org/index.php?title=Compiling_ScummVM/C%2B%2B11_Readiness) at the time we did tests.